src/widgets/SimpleTable.js

```diff
@@ -0,0 +1,186 @@
+/* eslint-disable react/no-unused-prop-types */
+/* eslint-disable react/forbid-prop-types */
+/**
+ * mSupply Mobile
+ * Sustainable Solutions (NZ) Ltd. 2019
+ */
+
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import { StyleSheet, FlatList, View, Text } from 'react-native';
+import { ROW_BLUE, APP_FONT_FAMILY, SUSSOL_ORANGE } from '../globalStyles/index';
+
+/**
+ * Simple table component only rendering scalar values in a cell. Not intended
+ * for entering data. Below are frequently used props. Many other props are
+ * available for styling, documented in PropTypes.
+ * @prop    {array}  data    Array of data. Example below.
+ * @prop    {array}  columns Array of column objects. Example below.
+ * @prop    {string} title   Title for this table.
+ * data = [ { a: 1, b: 2}, {a: 3, b: 4}, .. ]
+ * columns = [ { key: 'a', width: 1, title: 'A COLUMN' }, { key: 'b', width: 1, title: 'B COLUMN' }]
+ */
+export class SimpleTable extends React.PureComponent {
+  keyExtractor = ({ index }) => index;
+
+  renderTitle = () => {
+    const { title } = this.props;
+    const { titleContainer, titleFont } = localStyles(this.props);
+    return (
+      <View style={titleContainer}>
+        <Text style={titleFont}>{title}</Text>
+      </View>
+    );
+  };
+
+  renderCell = ({ content, index, header, width }) => {
+    const { cell, cellFont } = localStyles({ ...this.props, index, header, width });
+    return (
+      <View style={cell} key={this.keyExtractor({ index })}>
```

Comment:
https://reactjs.org/docs/lists-and-keys.html#extracting-components-with-keys

Reply:
from what i understand, it's should be enough to just pass key extractor to list, don't need them on cells.

Also you should be able to pass keyExtractor to this page (can use index by default, but apparently it has some performance issues when sorting)

Reply:
The example given is 2D, just rows, whereas this is 3D, rows and cells. Basically whenever something is rendered from an array (rows and cells for that row) it needs a key. I also get key warnings without it.

Had a conversation with @Chris-Petty  about keys. Basically in the end couldn't find a better unique value to use

Added keyExtractor prop 9ca521998257d360c189f1ced9a62611d48fb7d9

Reply:
![Screen Shot 2019-04-18 at 2 00 30 PM](https://user-images.githubusercontent.com/26194949/56332113-23f15600-61e3-11e9-91ad-c3d47afde2bf.png)

I am not a react expert, but i am suspicious about having the same key for siblings. And since every cell in a row has the same key, that seems to be the fact here. btw, map also calls callback with 'index' as second optional parameter, so you can pass it to render cell as the key

Reply:
d07b0e478c4810b25deee727f559ecf2ec2d4fc6